src/engine/shared/snapshot.cpp

```diff
@@ -243,7 +243,8 @@ void CSnapshotDelta::UndiffItem(const int *pPast, const int *pDiff, int *pOut, i
 		{
 			unsigned char aBuf[CVariableInt::MAX_BYTES_PACKED];
 			unsigned char *pEnd = CVariableInt::Pack(aBuf, *pDiff, sizeof(aBuf));
-			*pDataRate += (int)(pEnd - (unsigned char *)aBuf) * 8;
+			// addition with wrapping by casting to unsigned
+			*pDataRate += (unsigned)(pEnd - (unsigned char *)aBuf) * 8;
```

Comment:
This doesn't work. The addition is still an `int` addition.
```suggestion
			*pDataRate = (unsigned)*pDataRate + (unsigned)(int)(pEnd - (unsigned char *)aBuf) * 8;
```

But we'd have to make sure that wrapping is actually the correct behavior first.

Reply:
It is not correct to wrap. But it only affects a debug counter. I do not think it is worth it to allocate more memory for it. So lets keep it at wrapping and just make sure that clang doesn't trip on it.